scenarios/aws/kindvm/run.go

```diff
@@ -39,9 +45,20 @@ func Run(ctx *pulumi.Context) error {
 		return err
 	}
 
+	// Install docker if not installed yet, we need it to configure docker credentials
+	_, dockerInstallCmd, err := docker.NewManager(*awsEnv.CommonEnvironment, vm)
```

Comment:
It's fine, I am wondering if we should have it inside the `NewKindCluster` component, as it needs it. Can be done later

Reply:
Would have been easier, but we do not want to put AWS specific logic in the `NewKindCluster` component. That could be used on any cloud provider

Reply:
We'll need to duplicate this code in `datadog-agent` with current code. However you could have a `WithECRCredentials` that implements this. Note the `ExtraMount` you're adding means that it will currently fail when not using any credential helper. You may need to always create a empty `{}` JSON file.

You also don't need `docker login`, what `docker login` does is very basic JSON gen. The way I see it the clean way would be to build a small `Go` code that imports https://github.com/awslabs/amazon-ecr-credential-helper/blob/main/ecr-login/ecr.go#L83-L110 and produces the `config.json` on `stdout`.

It's much smaller and faster to download then the AWS SDK and you have the option to package as a docker container and get the `config.json` with `docker run --network host <your_helper_image> > config-tmp.json`.